build-tools/cmake/xa_macros.cmake

```diff
@@ -200,7 +200,8 @@ function(xa_common_prepare)
     -fstack-protector-strong
     -fstrict-return
     -fno-strict-aliasing
-    -ffunction-sections
+    -fno-function-sections
```

Comment:
What does this *do*? Why add it?  From the [`clang -ffunction-sections` docs](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fno-function-sections):

> Place each function in its own section

I thus infer that `-fno-function-sections` *doesn't* place each function into its own section?  I guess that could make the `.so` smaller?

Likewise the following line, [`clang -fno-data-sections`](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fno-data-sections) implies that each data is *not* placed into its own section.

Reply:
In theory it can make the app/library smaller, but at a (slight) cost of speed of execution (functions might end up far from each other and force long instead of short calls, for instance) and it might impede LTO (link-time optimizations), so I opted for the "traditional" section sharing.